fe/fe-core/src/main/java/com/starrocks/connector/ConnectorPartitionTraits.java

```diff
@@ -435,6 +445,53 @@ public Optional<Long> maxPartitionRefreshTs() {
             IcebergTable icebergTable = (IcebergTable) table;
             return icebergTable.getSnapshot().map(Snapshot::timestampMillis);
         }
+
+        @Override
+        public PartitionKey createPartitionKey(List<String> values, List<Column> columns) throws AnalysisException {
+            Preconditions.checkState(values.size() == columns.size(),
+                    "columns size is %s, but values size is %s", columns.size(), values.size());
+
+            IcebergTable icebergTable = (IcebergTable) table;
+            List<PartitionField> partitionFields = Lists.newArrayList();
+            for (Column column : columns) {
```

Comment:
what if the table has thousands of columns

Reply:
This is partition columns, can not be thousands of columns, I have rename the argments name